File: optional/legacy-multistack/batch/batch-lambda/src/utils/logger.ts
type LogLevel = 'debug' | 'info' | 'warn' | 'error'

interface LogEntry {
  timestamp: string
  level: LogLevel
  message: string
  context?: Record<string, unknown>
}

function formatLog(level: LogLevel, message: string, context?: Record<string, unknown>): string {
  const entry: LogEntry = {
    timestamp: new Date().toISOString(),
    level,
    message,
    ...(context && { context }),
  }
  return JSON.stringify(entry)
}

export const logger = {
  debug: (message: string, context?: Record<string, unknown>) => {
    console.debug(formatLog('debug', message, context))
  },

  info: (message: string, context?: Record<string, unknown>) => {
    console.log(formatLog('info', message, context))
  },

  warn: (message: string, context?: Record<string, unknown>) => {
    console.warn(formatLog('warn', message, context))
  },

  error: (message: string, context?: Record<string, unknown>) => {
    console.error(formatLog('error', message, context))
  },
}

File: optional/legacy-multistack/batch/batch-ecs-run/utils/logger.py
import json
from datetime import datetime, timezone
from typing import Any


class Logger:
    """構造化ログユーティリティ（Lambda/ECS統一フォーマット）"""

    def _format(self, level: str, message: str, context: dict[str, Any] | None = None) -> str:
        entry = {
            "timestamp": datetime.now(timezone.utc).isoformat(),
            "level": level,
            "message": message,
        }
        if context:
            entry["context"] = context
        return json.dumps(entry, ensure_ascii=False, default=str)

    def debug(self, message: str, context: dict[str, Any] | None = None) -> None:
        print(self._format("debug", message, context))

    def info(self, message: str, context: dict[str, Any] | None = None) -> None:
        print(self._format("info", message, context))

    def warn(self, message: str, context: dict[str, Any] | None = None) -> None:
        print(self._format("warn", message, context))

    def error(self, message: str, context: dict[str, Any] | None = None) -> None:
        print(self._format("error", message, context))


logger = Logger()
